src/items.rs

```diff
@@ -2295,158 +2266,47 @@ fn rewrite_args(
             .trim();
         return Some(comment.to_owned());
     }
-
-    let mut arg_item_strs = args
-        .iter()
-        .map(|arg| {
+    let arg_items: Vec<_> = itemize_list(
+        context.snippet_provider,
+        args.iter(),
+        ")",
+        ",",
+        |arg| span_lo_for_arg(arg),
+        |arg| arg.ty.span.hi(),
+        |arg| {
             arg.rewrite(context, Shape::legacy(multi_line_budget, arg_indent))
-                .unwrap_or_else(|| context.snippet(arg.span()).to_owned())
-        })
-        .collect::<Vec<_>>();
-
-    // Account for sugary self.
-    let mut pre_comment_str = "";
-    let mut post_comment_str = "";
-    let min_args = explicit_self
-        .and_then(|explicit_self| rewrite_explicit_self(explicit_self, args, context))
-        .map_or(1, |self_str| {
-            pre_comment_str = context.snippet(mk_sp(span.lo(), args[0].pat.span.lo()));
```

Comment:
I thought this was needed for the fix of issue #3198 but obviously tests pass. Do you know why ?

Reply:
AFAIK an AST node corresponding to `self` used not to have `Span`, so rustfmt had to craft spans manually. However with current libsyntax the AST node for `self` comes with its `Span`, so we can delegate this work to `write_list`.

Reply:
ok thanks